Doesn't work on 32 bit machines

The package does not compile on 32 bit machines due to an integer overflow:
```
../gopath219386341/pkg/mod/github.com/pnelson/numerizer@v0.0.0-20190602175649-b39ceac12567/numerizer.go:51:14: constant 1000000000000 overflows int
```
Use `int64` rather than `int`...

Seems to work:
```
tim@ultra:~/projects/numerizer$ GOARCH=386 go build
# github.com/pnelson/numerizer
./numerizer.go:51:14: constant 1000000000000 overflows int
tim@ultra:~/projects/numerizer$ git checkout 32-bit 
Zu Branch '32-bit' gewechselt
Ihr Branch ist auf demselben Stand wie 'origin/32-bit'.
tim@ultra:~/projects/numerizer$ GOARCH=386 go build
tim@ultra:~/projects/numerizer$
```

--- a/numerizer.go
+++ b/numerizer.go
@@ -6,7 +6,7 @@ import (
 	"strings"
 )
 
-var singleNumbers = map[string]int{
+var singleNumbers = map[string]int64{
 	"one":   1,
 	"two":   2,
 	"three": 3,
@@ -18,7 +18,7 @@ var singleNumbers = map[string]int{
 	"nine":  9,
 }
 
-var directNumbers = map[string]int{
+var directNumbers = map[string]int64{
 	"ten":       10,
 	"eleven":    11,
 	"twelve":    12,
@@ -32,7 +32,7 @@ var directNumbers = map[string]int{
 	"ninteen":   19, // misspelling
 }
 
-var tenPrefixNumbers = map[string]int{
+var tenPrefixNumbers = map[string]int64{
 	"twenty":  20,
 	"thirty":  30,
 	"forty":   40,
@@ -44,20 +44,24 @@ var tenPrefixNumbers = map[string]int{
 	"ninety":  90,
 }
 
-var largeNumbers = map[string]int{
-	"thousand": 1000,
-	"million":  1000000,
-	"billion":  1000000000,
-	"trillion": 1000000000000,
+// supports up to max signed 64 bit integer
+// 2^63-1 = 9223372036854775807
+var largeNumbers = map[string]int64{
+	"thousand":    1000,
+	"million":     1000000,
+	"billion":     1000000000,
+	"trillion":    1000000000000,
+	"quadrillion": 1000000000000000,
+	"quintillion": 1000000000000000000,
 }
 
 type item struct {
 	typ itemType
 	key string
-	val int
+	val int64
 }
 
-type itemType int
+type itemType int64
 
 const (
 	itemError itemType = iota
@@ -106,8 +110,8 @@ func (i item) String() string {
 type parser struct {
 	items []item
 	pos   int
-	prev  int
-	sum   int
+	prev  int64
+	sum   int64
 	err   error
 }
 
@@ -143,7 +147,7 @@ func preprocess(s string) string {
 type parseFn func(*parser) parseFn
 
 // Parse parses the provided string to an integer.
-func Parse(s string) (int, error) {
+func Parse(s string) (int64, error) {
 	p := newParser(s)
 	for state := parse; state != nil; {
 		state = state(p)

--- a/numerizer_test.go
+++ b/numerizer_test.go
@@ -2,56 +2,58 @@ package numerizer
 
 import "testing"
 
-func TestParse(t *testing.T) {
-	var tests = []struct {
-		in   string
-		want int
-	}{
-		{"zero", 0},
-		{"one", 1},
-		{"three", 3},
-		{"eleven", 11},
-		{"seventeen", 17},
-		{"sixty", 60},
-		{"one hundred", 100},
-		{"two hundred", 200},
-		{"three thousand", 3000},
-		{"four million", 4000000},
-		{"five billion", 5000000000},
-		{"six trillion", 6000000000000},
-		{"two hundred four", 204},
-		{"two hundred and four", 204},
-		{"seventeen hundred", 1700},
-		{"three thousand four", 3004},
-		{"three thousand and four", 3004},
-		{"three thousand sixteen", 3016},
-		{"three thousand and sixteen", 3016},
-		{"three thousand thirty", 3030},
-		{"three thousand and thirty", 3030},
-		{"three thousand thirty three", 3033},
-		{"three thousand four hundred", 3400},
-		{"three thousand and four hundred", 3400},
-		{"three thousand five hundred one", 3501},
-		{"three thousand five hundred and one", 3501},
-		{"three thousand six hundred twelve", 3612},
-		{"three thousand six hundred and twelve", 3612},
-		{"three thousand six hundred eighty", 3680},
-		{"three thousand six hundred and eighty", 3680},
-		{"three thousand six hundred eighty four", 3684},
-		{"three thousand six hundred and eighty four", 3684},
-		{"ten thousand", 10000},
-		{"twenty thousand", 20000},
-		{"three hundred thousand", 300000},
-		{"forty five", 45},
-		{"forty five hundred", 4500},
-		{"forty five thousand", 45000},
-		{"nine hundred ninety nine thousand nine hundred ninety nine", 999999},
-		{"nine hundred ninety nine million nine hundred ninety nine thousand nine hundred ninety nine", 999999999},
-		{"forty-five", 45},
-		{"four thousand, four hundred", 4400},
-		{"four thousand, four hundred thirty-two", 4432},
-	}
+var tests = []struct {
+	in   string
+	want int64
+}{
+	{"zero", 0},
+	{"one", 1},
+	{"three", 3},
+	{"eleven", 11},
+	{"seventeen", 17},
+	{"sixty", 60},
+	{"one hundred", 100},
+	{"two hundred", 200},
+	{"three thousand", 3000},
+	{"four million", 4000000},
+	{"five billion", 5000000000},
+	{"six trillion", 6000000000000},
+	{"eight quadrillion", 8000000000000000},
+	{"nine quintillion two hundred twenty-three quadrillion three hundred seventy-two trillion thirty-six billion eight hundred fifty-four million seven hundred seventy-five thousand eight hundred seven", 9223372036854775807},
+	{"two hundred four", 204},
+	{"two hundred and four", 204},
+	{"seventeen hundred", 1700},
+	{"three thousand four", 3004},
+	{"three thousand and four", 3004},
+	{"three thousand sixteen", 3016},
+	{"three thousand and sixteen", 3016},
+	{"three thousand thirty", 3030},
+	{"three thousand and thirty", 3030},
+	{"three thousand thirty three", 3033},
+	{"three thousand four hundred", 3400},
+	{"three thousand and four hundred", 3400},
+	{"three thousand five hundred one", 3501},
+	{"three thousand five hundred and one", 3501},
+	{"three thousand six hundred twelve", 3612},
+	{"three thousand six hundred and twelve", 3612},
+	{"three thousand six hundred eighty", 3680},
+	{"three thousand six hundred and eighty", 3680},
+	{"three thousand six hundred eighty four", 3684},
+	{"three thousand six hundred and eighty four", 3684},
+	{"ten thousand", 10000},
+	{"twenty thousand", 20000},
+	{"three hundred thousand", 300000},
+	{"forty five", 45},
+	{"forty five hundred", 4500},
+	{"forty five thousand", 45000},
+	{"nine hundred ninety nine thousand nine hundred ninety nine", 999999},
+	{"nine hundred ninety nine million nine hundred ninety nine thousand nine hundred ninety nine", 999999999},
+	{"forty-five", 45},
+	{"four thousand, four hundred", 4400},
+	{"four thousand, four hundred thirty-two", 4432},
+}
 
+func TestParse(t *testing.T) {
 	for _, tt := range tests {
 		have, err := Parse(tt.in)
 		if err != nil {
@@ -70,9 +72,7 @@ func TestParseError(t *testing.T) {
 		"hundred",
 		"thousand",
 		"zero four",
-		"four zero",
-		"five three",
-		"four eight",
+		"four zero", "five three", "four eight",
 		"eight seventy",
 		"eleven six",
 		"twelve seventeen",
@@ -85,11 +85,27 @@ func TestParseError(t *testing.T) {
 		"forty five thousand and",
 		"two and three",
 	}
-
 	for _, tt := range tests {
 		have, err := Parse(tt)
 		if err == nil {
 			t.Errorf("Parse(%q)\nhave %d\nwant parse error", tt, have)
 		}
 	}
+}
+
+// avoids compiler optimization
+var bench int64
+
+func BenchmarkParse(b *testing.B) {
+	var n int64
+	var err error
+	for i := 0; i < b.N; i++ {
+		for _, tt := range tests {
+			n, err = Parse(tt.in)
+			if err != nil {
+				b.Fatalf("Parse(%q) %v", tt.in, err)
+			}
+		}
+	}
+	bench = n
 }

